boa/src/builtins/mod.rs

```diff
@@ -22,30 +22,42 @@ pub(crate) use self::{
     bigint::BigInt,
     boolean::Boolean,
     error::{Error, RangeError, TypeError},
-    function::Function,
+    json::Json,
+    math::Math,
+    nan::NaN,
     number::Number,
     regexp::RegExp,
     string::String,
+    symbol::Symbol,
     value::{ResultValue, Value},
 };
 
 /// Initializes builtin objects and functions
 #[inline]
 pub fn init(global: &Value) {
```

Comment:
I think this should be mostly easy to implement:
```suggestion
pub fn init(global: &mut Object) {
```

We would need to use `Object::default()` in the Realm constructor [here](https://github.com/boa-dev/boa/blob/master/boa/src/realm.rs#L35-L54). And then implement `From<Object>` for `Value`.

Finally, I would change the `new_object()` function [here](https://github.com/boa-dev/boa/blob/73d7a64bb4da6040e3424297082d47debb3c91b8/boa/src/builtins/value/mod.rs#L135-L145) to not accept an `Option`, but to require always a global object.

This would allow implementing initialization with a very fast implementation, even doing it multi-threaded at some point.

Then, I think that we should create a trait or a type with a function that receives a `&mut Object` as a parameter and returns a `(&'static str, Value)`, this would help with creating an external API.

Reply:
yes. this would make somethings easier but what if we wanted an object to reference the global object (for example `globalThis`), this would make it really hard, right?

Reply:
> yes. this would make somethings easier but what if we wanted an object to reference the global object (for example `globalThis`), this would make it really hard, right?

Hmmm yes, this would be more difficult, like if we want to have `top/window/globalThis`, all pointing to the global object.

I think we can merge it as it is now, and maybe think on a design in the future :)